data/account/participationRegistry.go

```diff
@@ -366,11 +374,12 @@ func (db *participationDB) writeThread() {
 	for {
 		var wr partDBWriteRecord
 		var chanOk bool
-		// A call to .Flush() waits until the flash command passes the command queue, and also that the command queue is empty.
+		// A call to .Flush() waits until the flush command passes the command queue, and also that the command queue is empty.
 		if needsFlush {
 			select {
 			case wr, chanOk = <-db.writeQueue:
 				if !chanOk {
+					close(db.writeQueueDone)
```

Comment:
it's correct to place it here, but would be cleaner to use a
```golang
defer close(db.writeQueueDone)
```
at the beginning of the method; that way, you make sure that whenever the method exists the channel is being closed.